container-disc/src/main/java/com/yahoo/container/jdisc/DataplaneProxyService.java

```diff
@@ -235,6 +240,30 @@ static String nginxConfig(
                     .collect(Collectors.joining("\n"));
             nginxTemplate = replace(nginxTemplate, "vespa_token_endpoints", tokenmapping);
 
+            String corsMap = isDevEnvironment ? """
+                    map $http_origin $allow_origin {
+                            ~^https://.*\\.vespa-cloud.com$ $http_origin;
+                            ~^https?://localhost(:\\d+)?$ $http_origin;
+                            default "";
+                        }
+                    """ : "";
+
+            String corsHandler = isDevEnvironment ? """
+                    if ($request_method = OPTIONS) {
+                                  add_header Access-Control-Allow-Origin $allow_origin;
+                                  add_header Vary Origin;
+                                  add_header Access-Control-Allow-Headers "Authorization";
+                                  add_header Content-Length 0;
+                                  add_header Content-Type text/plain;
+                                  return 200 "";
+                                }
+                                add_header Access-Control-Allow-Origin $allow_origin;
+                                add_header Vary Origin;
```

Comment:
Afaik the `Access-Control-Request-Method` response header is required for pre-flight requests